### PR TITLE
Fix subterranean units teleporting to surface and jump jets to ground

### LINKED ISSUE #18731: TS: Jumpjet infantry immediately jumps down without transition
Jumpjet infantry only properly transitions to the air, towards ground it seems to change the layer immediately.

### PR DESCRIPTION
**Pull request by PunkPun**
Closes #18731

#17990 broke a lot of similar scenarios and #18321 fixed most of them, only this scenario wasn't accounted in the fix 

Move activity sticks unit CenterPosition straight to ground layer, if at ground layer. And it knows which layer a unit is via occupies space TopLeft => ToCell. Meaning because the subterranean unit is traveling to a cell at layer 0, it considers the unit as being on layer 0, regardless of it currently still being underground.

The implementation of #17990 seems a bit bogus, so I added a HACK comment for the future

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 1]
```diff
@@ -362,9 +363,10 @@ abstract class MovePart : Activity
 			protected readonly int Distance;
 			readonly int terrainOrientationMargin;
 			protected int progress;
+			protected bool movingOnGroundLayer;
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: `readonly`

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 2]
```diff
@@ -512,7 +517,8 @@ protected override MovePart OnComplete(Actor self, Mobile mobile, Move parent)
 							ToTerrainOrientation,
 							nextToTerrainOrientation,
 							margin,
-							progress - Distance);
+							progress - Distance,
+							mobile.ToCell.Layer == 0 && nextCell.Value.Cell.Layer == 0);
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Activities/Move/Move.cs:521]: For my own understanding, we have `FromCell`, `ToCell`, `nextCell` in scope here. Why are only the latter two important for this check?

**Review by RoosterDragon — COMMENTED** [review]
Working nicely ingame.

File: OpenRA.Mods.Common/Activities/Move/Move.cs (hunk 2 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/Move/Move.cs:521]: See `.SetLocation` a few lines below. We care about how we are going to move into the next cell

Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/Move/Move.cs:521]: FromCell is redundant here, as we have already passed it

**Review by RoosterDragon — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e4543de039e92fb9c9f0567b2d9116cd699cbda6...15dc7935a1c41353e0be36bca4ea7266c9690db7)

**Pull request merged**